runtime/lua/vim/lsp/_folding_range.lua

```diff
@@ -197,6 +197,22 @@ local function setup(bufnr)
     -- `on_detach` also runs on buffer reload (`:e`).
     -- Ensure `bufstate` and hooks are cleared to avoid duplication or leftover states.
     on_detach = function()
+      for _, client in
+        ipairs(vim.lsp.get_clients({
+          bufnr = bufnr,
+          method = ms.textDocument_foldingRange,
+        }))
+      do
+        for id, req in pairs(client.requests) do
```

Comment:
Even if we decide to only do this for folding, it seems worth lifting this into a `lsp.util._cancel_all_requests(method)` util function?

here is another case: https://github.com/neovim/neovim/blob/92e61072ac7493d85c15e3403f17619446782b51/runtime/lua/vim/lsp/util.lua#L2164-L2169